cartographer/mapping/detect_floors.cc

```diff
@@ -0,0 +1,204 @@
+/*
+ * Copyright 2016 The Cartographer Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "cartographer/mapping/detect_floors.h"
+
+#include <algorithm>
+#include <fstream>
+#include <vector>
+
+#include "Eigen/Core"
+#include "cartographer/transform/transform.h"
+#include "glog/logging.h"
+
+namespace cartographer {
+namespace mapping {
+
+namespace {
+
+// A union-find structure for assigning levels to 'spans' in the trajectory.
+using Levels = std::map<int, int>;
```

Comment:
Could you elaborate on the comment here a bit? What do the keys and values represent?